### ISSUE TITLE
[BUG] <title>token截断导致训练label被删除

### ISSUE THREAD

**Issue by tong-1989** [OpenBMB/MiniCPM-V#662]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

![img](https://github.com/user-attachments/assets/58f0cd0e-38ec-4ae6-bff8-638e19209a52)
在做finetune的时候，dataset.py下data_collator方法，会对数据做截断，丢掉max_length之后的tokens。这样在做监督训练的会后，会把模型学习的label截掉，导致报错"No tokens available to compute loss."

### 期望行为 | Expected Behavior

是否在做token截断的时候只对input做，不影响label的获取

### 运行环境 | Environment

```Markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):
```

**Comment by LDLINGLINGLING** [comment]
你好，个人认为这里很难改，因为不好同时兼顾自动判断有效token长度和个人的显存。

**Comment by DankoZhang** [comment]
用swift框架吧，直接删除超长的数据
